chall04/vde-dios.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+import sys
+import re
+
+
+def error(e , i):
+    file_name =  "stdin" if i == 0 else sys.argv[i]
+    if (e == 1):
+        print("%s: %s: Bad format" %(sys.argv[0], file_name))
+    elif (e == 2):
+        print("%s: %s: Can't read file" %(sys.argv[0], file_name))
+    elif (e == 3):
+        print("%s: %s: Not enough space in the given shelves" %(sys.argv[0], file_name))
+
+
+# Check format plus sort shelves
+def shelf_parse(shelves):
+    l = shelves.replace("\n", "").split(" ")
```

Comment:
You do not need to replace the `\n`s if you use split without parameters:
```suggestion
    l = shelves.split()
```
As stated in the [method's documentation](https://docs.python.org/3/library/stdtypes.html#str.split):
> If sep is not specified or is None, a different splitting algorithm is applied: runs of consecutive whitespace are regarded as a single separator, and the result will contain no empty strings at the start or end if the string has leading or trailing whitespace. Consequently, splitting an empty string or a string consisting of just whitespace with a None separator returns [].